### PR TITLE
Adds Compiler

### PR DESCRIPTION
**Pull request by btamadio**
Adds a compiler for turning PlanOut scripts into code that can be executed by the Interpreter.

See the README update for more details

### CODE REVIEW

File: compiler/ast/ast.go [hunk 1]
```diff
@@ -0,0 +1,383 @@
+package ast
+
+import (
+	"github.com/stitchfix/planout-golang/compiler/token"
```
Comment (GregBowyer) [inline compiler/ast/ast.go]: I would assume these need changing right?

**Review by GregBowyer — CHANGES_REQUESTED** [review]
It looks fine to me, I might rebase / squash the commits together.

Do the imports need to change?

File: compiler/ast/ast.go (hunk 1 above)
Comment (btamadio) [reply compiler/ast/ast.go]: Just pushed the changes to rename the imports

**Comment by btamadio** [comment]
Just following up on this - imports have been updated as requested! Would love to get this merged if possible!

**Comment by btamadio** [comment]
> It looks fine to me, I might rebase / squash the commits together.
> 
> Do the imports need to change?

Imports have been updated. Any chance this could get merged?

**Pull request merged**